### PR TITLE
Improved time control UI 

### PR DESCRIPTION
**Pull request by sunreef**
The pause button now sends an input action called `TogglePause`, just like the P key. Pressing the button again or pressing P again will send the same `TogglePause` action to resume the game. 

I also added `Speed Up` and `Slow Down`  buttons to control the simulation speed.

### CODE REVIEW

File: src/states/main_game.rs [hunk 1]
```diff
@@ -213,6 +226,33 @@ impl SimpleState for MainGameState {
...
+                    }
+                }
+                _ => (),
+            }
+        }
         Trans::None
     }
+
+    fn on_resume(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        {
```
Comment (marotili) [inline src/states/main_game.rs]: Could you add comments on both `on_resume` functions as to why you are re-registering the reader?

File: src/states/paused.rs [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
+        {
+            let mut input_event_channel = data
+                .world
+                .write_resource::<EventChannel<InputEvent<String>>>();
+            self.input_event_reader_id = Some(input_event_channel.register_reader());
         }
+    }
+
+    fn on_resume(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        {
```
Comment (marotili) [inline src/states/paused.rs]: There are a couple of unnecessary scopes, could you remove them?

Comment (sunreef) [reply src/states/paused.rs]: Should be OK now.

Comment (marotili) [reply src/states/paused.rs]: I meant the additional braces in line `21` and `26` - but it's good to know that it is also possible to remove the lifetime definitions :)

Comment (sunreef) [reply src/states/paused.rs]: Oh, the additional braces are here on purpose. If we want to use the `data.world` for something else later on, we need them to end the mutable borrow.

Comment (sunreef) [reply src/states/paused.rs]: Although I can just rewrite it so that the mutable borrow ends directly.

Comment (marotili) [reply src/states/paused.rs]: Good point, always code with change in mind 🙏

**Review by marotili — APPROVED** [review]

**Pull request merged**